src/buildercore/terraform.py

```diff
@@ -564,6 +569,230 @@ def render_bigquery(context, template):
+        'policy_arn': "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+        'role': "${aws_iam_role.worker.name}",
+    })
+
+    template.populate_resource('aws_iam_role_policy_attachment', 'worker_cni', block={
+        'policy_arn': "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+        'role': "${aws_iam_role.worker.name}",
+    })
+
+    template.populate_resource('aws_iam_role_policy_attachment', 'worker_ecr', block={
+        'policy_arn': "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+        'role': "${aws_iam_role.worker.name}",
+    })
+
+    template.populate_resource('aws_iam_instance_profile', 'worker', block={
+        'name': '%s--worker' % context['stackname'],
+        'role': '${aws_iam_role.worker.name}'
+    })
+
+    template.populate_data(DATA_TYPE_AWS_AMI, 'worker', block={
+        'filter': {
+            'name': 'name',
+            'values': ['amazon-eks-node-%s-v*' % context['eks']['version']],
+        },
+        'most_recent': True,
+        'owners': [aws.ACCOUNT_EKS_AMI],
+    })
+
+    # EKS currently documents this required userdata for EKS worker nodes to
+    # properly configure Kubernetes applications on the EC2 instance.
+    # We utilize a Terraform local here to simplify Base64 encoding this
+    # information into the AutoScaling Launch Configuration.
+    # More information: https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html
+    template.populate_local('worker_userdata', """
+#!/bin/bash
+set -o xtrace
+/etc/eks/bootstrap.sh --apiserver-endpoint '${aws_eks_cluster.main.endpoint}' --b64-cluster-ca '${aws_eks_cluster.main.certificate_authority.0.data}' '${aws_eks_cluster.main.name}'""")
+
+    template.populate_resource('aws_launch_configuration', 'worker', block={
+        'associate_public_ip_address': True,
+        'iam_instance_profile': '${aws_iam_instance_profile.worker.name}',
+        'image_id': '${data.aws_ami.worker.id}',
+        'instance_type': context['eks']['worker']['type'],
+        'name_prefix': '%s--worker' % context['stackname'],
+        'security_groups': ['${aws_security_group.worker.id}'],
+        'user_data_base64': '${base64encode(local.worker_userdata)}',
+        'lifecycle': {
+            'create_before_destroy': True,
+        },
+    })
+
+    autoscaling_group_tags = [{
+        'key': k,
+        'value': v,
+        'propagate_at_launch': True,
+    } for (k, v) in aws.generic_tags(context).items()]
+    autoscaling_group_tags.append({
+        'key': 'kubernetes.io/cluster/%s' % context['stackname'],
+        'value': 'owned',
+        'propagate_at_launch': True,
+    })
+    template.populate_resource('aws_autoscaling_group', 'worker', block={
+        'name': '%s--worker' % context['stackname'],
+        'launch_configuration': '${aws_launch_configuration.worker.id}',
+        'min_size': context['eks']['worker'].get('min-size', 1),
```

Comment:
Here we require `eks` and `worker` but `min-size` is optional with a default. Effectively we have the same default value we configured in `elife.yaml` `defaults`. I don't know how the configuration merging works, I guess it doesn't do nested merging?

Reply:
We shouldn't really need this level of defaults if the `elife.yaml` (or its dummy for testing) are well-formed, so I'll remove.